### PR TITLE
Adds support for ChunkWM

### PR DESCRIPTION
**Pull request by peel**
As KWM is now deprecated and ChunkWM is the thing to follow, I made myself a module for configuring it. 
The quirk however is that as of now [ChunkWM does not support custom config path](https://github.com/koekeishiya/chunkwm/issues/96), hence you either need to:
- make `/etc/{chunkwmrc,chunkwmtilingrc}` executable and symlink `/etc/{chunkwmrc,chunkwmtilingrc}` as $HOME dotfiles
- create executable wrappers in $HOME sourcing the above (as I'm using stow to manage dotfiles that's what I do for now)

Sidenote: I also do have ugly derivations for installing ChunkWM and plugins if you're interested

### CODE REVIEW

File: modules/services/chunkwm.nix [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+    services.chunkwm.port = mkOption {
+      type = types.int;
+      default = 3920;
+      description = "ChunkC port";
+    };
+
+    services.chunkwm.plugins.dir = mkOption {
+      type = types.path;
+      default = "/run/current-system/sw/lib/chunkwm/plugins";
```
Comment (peel) [inline modules/services/chunkwm.nix]: not sure about the default though

File: modules/services/chunkwm.nix [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      description = "Chunkwm Plugins directory";
+    };
+
+    services.chunkwm.plugins.list = mkOption {
+      type = types.listOf (types.enum [ "border" "tiling" "ffm" ]);
+      default = [ "border" "tiling" "ffm" ];
```
Comment (LnL7) [inline modules/services/chunkwm.nix]: list types are appended by default, moving these defaults to config means that people can add plugins without loosing the defaults.

File: modules/services/chunkwm.nix (hunk 1 above)
Comment (LnL7) [inline modules/services/chunkwm.nix]: nixos uses `/var/lib` for application data, but for store paths I think `share/chunkwm` makes more sense.

Comment (peel) [inline modules/services/chunkwm.nix]: Had some issues building chunkwm plugins and loading from share. Can we leave it for now?

Comment (LnL7) [inline modules/services/chunkwm.nix]: To clarify, I was talking about `/run/current-system/sw/share/chunkwm/plugins`. But it's fine if that doesn't work

Comment (peel) [inline modules/services/chunkwm.nix]: I'm following closely the work being done on the project. Will move those to /share once it's stable-ish.

**Comment by LnL7** [comment]
Could you post the expression for chunkwm somewhere, then I can try it this out 😄 .

**Comment by peel** [comment]
Sure, it's kinda ugly for now, but will sort it out as the development of the WM is going along nicely ;)
https://github.com/peel/dotfiles/tree/master/nix/.nixpkgs/pkgs/os-specific/darwin/chunkwm

**Pull request merged**